awx/main/models/credential/__init__.py

```diff
@@ -430,6 +430,16 @@ def unique_dict(cred_qs):
             ret[cred.unique_hash()] = cred
         return ret
 
+    def get_input(self, field_name, **kwargs):
+        if field_name in self.credential_type.secret_fields:
+            return decrypt_field(self, field_name)
+        if 'default' in kwargs:
+            return getattr(self, field_name, kwargs['default'])
+        return getattr(self, field_name)
```

Comment:
I'm not a fan of using this as the default. If it's not an input, then I would prefer that it return None or raise an error or something. This can return unintended values or have side effects.

Reply:
Thats what this line does. Unless a default is passed you'll get an `AttributeError`. As I mentioned above the `default` behavior is more about being consistent with existing access patterns and avoiding inadvertently adding bugs.

Reply:
> This can return unintended values or have side effects.

I am not sure i follow :\ What sort of side-effects do are you referring to? Can you give an example?

Reply:
> If that raises an AttributeError, then yes, it will hand it back off to the python getattr logic.

Correct. That is the expected behavior.

Reply:
> processing encryption via one method but not the other

Correct - as I will be highlighting in the documentation string for this function, `get_input` absolutely will be something that:

1) dencrypts the return value if that input is a secret
2) potentially talks to a secret backend.

Can you elaborate on what you mean by 'more concerning'? What specifically is concerning?

Reply:
@AlanCoding Thanks for the responses - I've left some comments and examples for them. Please let me know what you think!